File: docs-examples/example-java/src/test/java/io/micronaut/rabbitmq/docs/exchange/AnimalClient.java
package io.micronaut.rabbitmq.docs.exchange;

import io.micronaut.context.annotation.Requires;
// tag::imports[]
import io.micronaut.messaging.annotation.MessageHeader;
import io.micronaut.rabbitmq.annotation.RabbitClient;
// end::imports[]

@Requires(property = "spec.name", value = "CustomExchangeSpec")
// tag::clazz[]
@RabbitClient("animals") // <1>
public interface AnimalClient {

    void send(@MessageHeader String animalType, Animal animal); // <2>

    default void send(Animal animal) { // <3>
        send(animal.getClass().getSimpleName(), animal);
    }
}
// end::clazz[]

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/rabbitmq/docs/exchange/AnimalClient.kt
package io.micronaut.rabbitmq.docs.exchange

import io.micronaut.context.annotation.Requires
// tag::imports[]
import io.micronaut.messaging.annotation.MessageHeader
import io.micronaut.rabbitmq.annotation.RabbitClient
// end::imports[]

@Requires(property = "spec.name", value = "CustomExchangeSpec")
// tag::clazz[]
@RabbitClient("animals") // <1>
abstract class AnimalClient {

    abstract fun send(@MessageHeader animalType: String, animal: Animal)  // <2>

    fun send(animal: Animal) { //<3>
        send(animal.javaClass.simpleName, animal)
    }
}
// end::clazz[]
